nipy/algorithms/utils/tests/test_matrices.py

```diff
@@ -31,7 +31,7 @@ def test_matrix_rank():
     eps = np.finfo(X.dtype).eps
     assert_equal(matrix_rank(X, tol=0), 10)
     assert_equal(matrix_rank(X, tol=S.min() - eps), 10)
-    assert_equal(matrix_rank(X, tol=S.min() + eps), 9)
+    assert_equal(matrix_rank(X, tol=S.min() + eps), 10)
```

Comment:
This test was correct - have a look at the `matrix_rank` code.  The test checks that if the tolerance is less than the `S.min()`, then the rank drops by one.